### PR TITLE
Hook up file watchers for analyzer references

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Continuation of https://github.com/dotnet/roslyn/pull/74745.  This ensures the initial hookup happens to cause us to listen to file changes for analyzer references on disk.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs [hunk 1]
```diff
@@ -653,12 +654,21 @@ await _projectSystemProjectFactory.ApplyBatchChangeToWorkspaceMaybeAsync(useAsyn
                         newSolution: solutionChanges.Solution.RemoveProjectReference(Id, projectReference));
                 }
 
+                // Analyzer reference removing...
+                {
+                    projectUpdateState = projectUpdateState.WithIncrementalAnalyzerReferencesRemoved(_analyzersRemovedInBatch);
+
+                    foreach (var analyzerReference in _analyzersRemovedInBatch)
+                        solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.RemoveAnalyzerReference(Id, analyzerReference));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs:662]: doesn't look exactly the same as below as we don't hav ea Solution.RemoveAnalyzerReferences (plural) to go with Solution.AddAnalyzerReferences.  And i'm not interested in adding that at this point :)

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs [hunk 2]
```diff
@@ -653,12 +654,21 @@ await _projectSystemProjectFactory.ApplyBatchChangeToWorkspaceMaybeAsync(useAsyn
...
                 // Analyzer reference adding...
-                solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.AddAnalyzerReferences(Id, _analyzersAddedInBatch));
+                {
+                    projectUpdateState = projectUpdateState.WithIncrementalAnalyzerReferencesAdded(_analyzersAddedInBatch);
 
-                // Analyzer reference removing...
-                foreach (var analyzerReference in _analyzersRemovedInBatch)
-                    solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.RemoveAnalyzerReference(Id, analyzerReference));
+                    solutionChanges.UpdateSolutionForProjectAction(
+                        Id, solutionChanges.Solution.AddAnalyzerReferences(Id, projectUpdateState.AddedAnalyzerReferences));
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs:670]: >projectUpdateState.AddedAnalyzerReferences) [](http://example.com/codeflow?start=79&length=43)

nit: this is also just _analyzersAddedInBatch, right? Seems a bit odd that the removing one uses _analyzersRemovedInBatch for it's solution update, whereas the adding on uses projectUpdateState.AddedAnalyzerReferences

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs [hunk 3]
```diff
@@ -653,12 +654,21 @@ await _projectSystemProjectFactory.ApplyBatchChangeToWorkspaceMaybeAsync(useAsyn
                         newSolution: solutionChanges.Solution.RemoveProjectReference(Id, projectReference));
                 }
 
+                // Analyzer reference removing...
+                {
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs:658]: >{ [](http://example.com/codeflow?start=16&length=1)

nit: might be nice to add a Count > 0 check for both these scopes, just to make it very evident that we're not doing anything in those cases.

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs:658]: sure.

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs]: ick.  yes.  i will make consistent.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun  can you please takea nother look.  @jasonmalinowski  as well.

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs [hunk 4]
```diff
@@ -918,47 +930,54 @@ public void AddAnalyzerReference(string fullPath)
             foreach (var mappedFullPath in mappedPaths)
             {
                 if (_analyzerPathsToAnalyzers.ContainsKey(mappedFullPath))
-                {
                     throw new ArgumentException($"'{fullPath}' has already been added to this project.", nameof(fullPath));
-                }
             }
 
-            foreach (var mappedFullPath in mappedPaths)
+            if (_activeBatchScopes > 0)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs:936]: i missed from this point downwards.

**Pull request merged**